sycl/include/sycl/ext/oneapi/kernel_properties/properties.hpp

```diff
@@ -95,9 +95,6 @@ struct property_value<work_group_size_hint_key,
 template <uint32_t Size>
 struct property_value<sub_group_size_key,
                       std::integral_constant<uint32_t, Size>> {
-  static_assert(Size != 0,
```

Comment:
I understand that we need it if we want to re-use existing `sub_group_size` property, but at the same time we are now allowing our users to set sub-group size to zero, which could be viewed as a user experience setback.

I don't think that it is of Critical or High importance,  though, because [`sycl_ext_oneapi_kernel_properties`](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_properties.asciidoc) does not exactly says that the argument can't be zero and SYCL 2020 spec [doesn't even say](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:kernel.attributes) that it can't be a negative value

Reply:
An alternative is to do similar to `automatic` and simply give it value `max - 1`, then specialize it to set the value in the generated IR to 0. That way we get to keep this limitation, although it means another magic value the users should not explicitly use. Hopefully there's never a use-case for a 4294967295 or 4294967294 size sub-group. 😉

Reply:
I still think this would be good to address. Using 0 as a magic value seems more likely to be hit by users than max and max-1.

Reply:
Added in [3dc8731](https://github.com/intel/llvm/pull/12335/commits/3dc87319fdc2a3cdc200c240c461144a76448b7b). I'll need to update the handling in translator, as 0 was treated as the special value to signal adding the named sub group size execution mode to the SPIRV module, but now that should be 4294967295 (= -1).